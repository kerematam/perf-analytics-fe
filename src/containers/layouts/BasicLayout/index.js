import React, { lazy, Suspense } from 'react'
import { Switch, Route, Redirect } from 'react-router-dom'
import Header from 'containers/layouts/Header'

import styles from './BasicLayout.module.scss'

const HomePage = lazy(() => import('pages/HomePage'))
const StatsPage = lazy(() => import('pages/StatsPage'))

const BasicLayout = () => {
  return (
    <div className={styles.root}>
      <Header />
      <div>
        <Suspense fallback={'Loading...'}>
          <Switch>
            <Redirect exact from="/" to="/home" />
            <Route path="/home" exact render={() => <HomePage />} />
            <Route path="/stats/:id" render={() => <StatsPage />} />
          </Switch>
        </Suspense>
      </div>
    </div>
  )
}

export default BasicLayout
