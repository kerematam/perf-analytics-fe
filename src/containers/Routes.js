import React from 'react'
import { Switch, Route, Redirect } from 'react-router-dom'
import BasicLayout from './layouts/BasicLayout'
import NotFoundPage from 'pages/NotFoundPage'

const useGetAuthenticated = () => {
  // TODO: change after authentication - katam
  const isAuthenticated = true

  return isAuthenticated
}

function ProtectedRoute({ children, isAuthenticated, ...rest }) {
  return (
    <Route
      {...rest}
      render={({ location }) =>
        isAuthenticated ? (
          children
        ) : (
          <Redirect
            to={{
              pathname: '/login',
              state: { from: location },
            }}
          />
        )
      }
    />
  )
}

function AuthRoute({ children, isAuthenticated, ...rest }) {
  return (
    <Route
      {...rest}
      render={({ location }) =>
        !isAuthenticated ? (
          children
        ) : (
          <Redirect
            to={{
              pathname: '/',
            }}
          />
        )
      }
    />
  )
}

const Routes = () => {
  const isAuthenticated = useGetAuthenticated()

  return (
    <Switch>
      <ProtectedRoute
        isAuthenticated={isAuthenticated}
        exact
        path={['/', '/home', '/stats/:id']}
      >
        <BasicLayout />
      </ProtectedRoute>
      <AuthRoute isAuthenticated={isAuthenticated} path={['/login']} />
      <Route path="/404" component={NotFoundPage} />
      <Redirect to="/404" />
    </Switch>
  )
}

export default Routes
