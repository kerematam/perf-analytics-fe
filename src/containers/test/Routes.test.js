import React from 'react'
import { Provider } from 'react-redux'
import { HelmetProvider } from 'react-helmet-async'
import configureStore from 'redux/configureStore'
import { MemoryRouter } from 'react-router-dom'
import { configure, mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import BasicLayout from '../layouts/BasicLayout'
import Routes from '../Routes'

configure({ adapter: new Adapter() })

const setup = (path = '/') => {
  const store = configureStore()

  const Component = (
    <Provider store={store}>
      <MemoryRouter initialEntries={[path]}>
        <HelmetProvider>
          <Routes />
        </HelmetProvider>
      </MemoryRouter>
    </Provider>
  )

  const snapshot = mount(Component).debug()
  const wrapper = mount(Component)

  return {
    snapshot,
    wrapper,
  }
}

describe('Routes should handle root path', () => {
  test('correctly handles "/" route', () => {
    const { wrapper, snapshot } = setup('/')

    expect(snapshot).toMatchSnapshot()
    expect(wrapper.find(BasicLayout).length).toBe(1)
  })

  test('correctly handles 404 routes ', () => {
    const { snapshot } = setup('/sdafadsfasdf')
    expect(snapshot).toMatchSnapshot()
  })
})
