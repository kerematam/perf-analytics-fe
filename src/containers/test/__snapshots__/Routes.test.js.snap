// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes should handle root path correctly handles "/" route 1`] = `
"<Provider store={{...}}>
  <MemoryRouter initialEntries={{...}}>
    <Router history={{...}}>
      <HelmetProvider context={{...}}>
        <Routes>
          <Switch>
            <ProtectedRoute isAuthenticated={true} exact={true} path={{...}} location={{...}} computedMatch={{...}}>
              <Route exact={true} path={{...}} location={{...}} computedMatch={{...}} render={[Function: render]}>
                <BasicLayout>
                  <div className=\\"root\\">
                    <LayoutHeader>
                      <div className=\\"root\\">
                        <span className=\\"menu_item\\">
                          <WithStyles(ForwardRef(Button)) onClick={[Function: handleClickHomepage]} className=\\"menu_item_button\\">
                            <ForwardRef(Button) classes={{...}} onClick={[Function: handleClickHomepage]} className=\\"menu_item_button\\">
                              <WithStyles(ForwardRef(ButtonBase)) className=\\"MuiButton-root MuiButton-text menu_item_button\\" component=\\"button\\" disabled={false} focusRipple={true} focusVisibleClassName=\\"Mui-focusVisible\\" type=\\"button\\" onClick={[Function: handleClickHomepage]}>
                                <ForwardRef(ButtonBase) classes={{...}} className=\\"MuiButton-root MuiButton-text menu_item_button\\" component=\\"button\\" disabled={false} focusRipple={true} focusVisibleClassName=\\"Mui-focusVisible\\" type=\\"button\\" onClick={[Function: handleClickHomepage]}>
                                  <button className=\\"MuiButtonBase-root MuiButton-root MuiButton-text menu_item_button\\" onBlur={[Function]} onClick={[Function: handleClickHomepage]} onFocus={[Function]} onKeyDown={[Function]} onKeyUp={[Function]} onMouseDown={[Function]} onMouseLeave={[Function]} onMouseUp={[Function]} onDragLeave={[Function]} onTouchEnd={[Function]} onTouchMove={[Function]} onTouchStart={[Function]} tabIndex={0} type=\\"button\\" disabled={false}>
                                    <span className=\\"MuiButton-label\\">
                                      <ForwardRef className=\\"menu_item_icon\\" fontSize=\\"small\\">
                                        <WithStyles(ForwardRef(SvgIcon)) className=\\"menu_item_icon\\" fontSize=\\"small\\">
                                          <ForwardRef(SvgIcon) classes={{...}} className=\\"menu_item_icon\\" fontSize=\\"small\\">
                                            <svg className=\\"MuiSvgIcon-root menu_item_icon MuiSvgIcon-fontSizeSmall\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" color={[undefined]} aria-hidden=\\"true\\" role=\\"presentation\\">
                                              <path d=\\"M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z\\" />
                                            </svg>
                                          </ForwardRef(SvgIcon)>
                                        </WithStyles(ForwardRef(SvgIcon))>
                                      </ForwardRef>
                                      <WithStyles(ForwardRef(Typography)) className=\\"menu_item_text\\">
                                        <ForwardRef(Typography) classes={{...}} className=\\"menu_item_text\\">
                                          <p className=\\"MuiTypography-root menu_item_text MuiTypography-body1\\">
                                            Home
                                          </p>
                                        </ForwardRef(Typography)>
                                      </WithStyles(ForwardRef(Typography))>
                                    </span>
                                    <NoSsr>
                                      <WithStyles(undefined) center={false}>
                                        <ForwardRef(TouchRipple) classes={{...}} center={false}>
                                          <span className=\\"MuiTouchRipple-root\\">
                                            <TransitionGroup component={{...}} exit={true} childFactory={[Function: childFactory]} />
                                          </span>
                                        </ForwardRef(TouchRipple)>
                                      </WithStyles(undefined)>
                                    </NoSsr>
                                  </button>
                                </ForwardRef(ButtonBase)>
                              </WithStyles(ForwardRef(ButtonBase))>
                            </ForwardRef(Button)>
                          </WithStyles(ForwardRef(Button))>
                        </span>
                      </div>
                    </LayoutHeader>
                    <div>
                      <Suspense fallback=\\"Loading...\\">
                        <Switch>
                          <Route path=\\"/home\\" exact={true} render={[Function: render]} location={{...}} computedMatch={{...}} />
                        </Switch>
                        Loading...
                      </Suspense>
                    </div>
                  </div>
                </BasicLayout>
              </Route>
            </ProtectedRoute>
          </Switch>
        </Routes>
      </HelmetProvider>
    </Router>
  </MemoryRouter>
</Provider>"
`;

exports[`Routes should handle root path correctly handles 404 routes  1`] = `
"<Provider store={{...}}>
  <MemoryRouter initialEntries={{...}}>
    <Router history={{...}}>
      <HelmetProvider context={{...}}>
        <Routes>
          <Switch>
            <Route path=\\"/404\\" component={[Function: NotFoundPage]} location={{...}} computedMatch={{...}}>
              <NotFoundPage history={{...}} location={{...}} match={{...}} staticContext={[undefined]}>
                <div className=\\"img_container\\">
                  <img alt=\\"404\\" src=\\"404.jpg\\" className=\\"img\\" />
                </div>
              </NotFoundPage>
            </Route>
          </Switch>
        </Routes>
      </HelmetProvider>
    </Router>
  </MemoryRouter>
</Provider>"
`;
